docs/dpdk/gnmi-ctl.rst

```diff
@@ -0,0 +1,288 @@
+gnmi-ctl executable
+-------------------
+
+gnmi-ctl is a gRPC-based C++ network management interface client to handle port configurations and program fixed functions in the P4 pipeline. This client connects
+to the Config monitoring service running as part of Stratum and integrated as part of
+infrap4d. This utility is built as part of networking recipe and is available under
+`install/bin/`.
+
+gnmi client connects to gRPC port 9339 opened by the server and sends a protobuf
+based message. Refer to gnmi.proto and gnmi_ext.proto for more details on type of
+messages and services that are available for a gNMI client.
+
+Each CLI format should be inline with the YANG parser available in server which
+is a tree based hierarchy. Tree has `interfaces` as root-node, followed by the
+device type. Each device can have multiple ports, where port-name acts as a KEY.
+SET command can be executed to configure the ports and GET command to retrieve the
+information about previously configured ports
+
+1) Set attributes for a vhost port::
+
+    $ gnmi-ctl set PARAMS
+    $ Example:
+    gnmi-ctl set "device:virtual-device,name:net_vhost0,device-type:VIRTIO_NET"
+    gnmi-ctl set "device:virtual-device,name:net_vhost0,port-type:LINK"
+    gnmi-ctl set "device:virtual-device,name:net_vhost0,host-name:host1,
+                  device-type:VIRTIO_NET,queues:1,
+                  socket-path:/tmp/vhost-user-0,port-type:LINK"
+    gnmi-ctl set "device:virtual-device,name:net_vhost0,host-name:host1,
+                  device-type:VIRTIO_NET,queues:1,
+                  pipeline-name:pipe,mempool-name:MEMPOOL0,mtu:2000,
+                  socket-path:/tmp/vhost-user-0,packet-dir:host,port-type:LINK"
+ ..
+
+* Parameter description:
+    * ``PARAMS``: These params are key:value pairs. Here virtual-device is a sub-node which can hold multiple ports like net_vhost0, net_vhost1,... and each port accepts multiple configuration parameters. These configuration parameters are again a key:value pair, which can be either passed in single CLI command or multiple CLI commands.
+
+    * ``name``: Supported values are defined in chassis config file `dpdk_port_config.pb.txt`.
+    * ``host-name``: can be any string.
+    * ``queues``: number of queues to be configured by target backend.
+    * ``socket-path``: socket path for vhost port.
+    * ``port-type``: Defines the port type and supported value for vhost port is `LINK`.
+    * ``device-type``: Defines the device type and supported value for vhost port is `VIRTIO_NET`.
+    * ``pipeline-name``: Defines the pipeline name, it's a non-mandatory parameter. Default value is `pipe`.
```

Comment:
Grammar error: comma splice (joining two independent clauses with a comma).
I'd also suggest _optional_ instead of _non-mandatory_.

In this context, you don't need to use complete sentences, so there are a number of terse alternatives.

1. Defines the pipeline name. Optional parameter. Default value is `pipe`.
2. Defines the pipeline name. Optional. Default value is `pipe`.
3. Defines the pipeline name. Default value is `pipe`. (optional)

This comment applies throughout the document.

Reply:
Will go with this option - Defines the pipeline name. Optional parameter. Default value is pipe.